shell/platform/embedder/embedder.h

```diff
@@ -434,6 +438,55 @@ typedef struct {
   BoolPresentInfoCallback present_with_info;
 } FlutterOpenGLRendererConfig;
 
+/// Alias for id<MTLDevice>.
+typedef const void* FlutterMetalDevice;
+
+/// Alias for id<MTLCommandQueue>.
+typedef const void* FlutterMetalCommandQueue;
+
+/// Alias for id<MTLTexture>.
+typedef const void* FlutterMetalTextureBuffer;
+
+typedef struct {
+  /// The size of this struct. Must be sizeof(FlutterMetalTexture).
+  size_t struct_size;
+  /// Embedder provided unique identifier to the texture buffer. Given that the
+  /// `texture` handle is passed to the engine to render to, the texture buffer
+  /// is itseld owned by the embedder. This `texture_id` is then also given to
+  /// the embedder in the present callback.
+  int64_t texture_id;
+  /// Handle to the MTLTexture that is owned by the embedder. Engine will render
+  /// the frame into this texture.
+  FlutterMetalTextureBuffer* texture;
+} FlutterMetalTexture;
+
+/// Callback for when a metal texture is requested.
+typedef FlutterMetalTexture (*FlutterMetalTextureCallback)(
+    void* /* user data */,
+    const FlutterFrameInfo* /* frame info */);
+
+/// Callback for when a metal texture is presented. The texture_id here
+/// corresponds to the texture_id provided by the embedder in the
+/// `FlutterMetalTextureCallback` callback.
+typedef bool (*FlutterMetalPresentCallback)(
+    void* /* user data */,
+    const FlutterMetalTexture* /* texture */);
```

Comment:
Again, a pointer to a pointer. Those are usually only required for out parameters.

Reply:
I think you missed this double pointer indirection.

Reply:
@chinmaygarde this isn't a handle. This is just a pointer to a struct.